prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -339,50 +364,88 @@ public APIResult submitAndSchedule(
-                feeds.add(entity.getName());
-            } else if (EntityType.PROCESS.equals(entity.getEntityType())) {
-                processes.add(entity.getName());
+    private List<Feed> getFeeds(List<FormDataBodyPart> feedForms) {
+        List<Feed> feeds = new ArrayList<>();
+        if (feedForms != null && !feedForms.isEmpty()) {
+            for (FormDataBodyPart formDataBodyPart : feedForms) {
+                feeds.add(formDataBodyPart.getValueAs(Feed.class));
+            }
+        }
+        return feeds;
+    }
+
+    private List<Process> getProcesses(List<FormDataBodyPart> processForms) {
+        List<Process> processes = new ArrayList<>();
+        if (processForms != null && !processForms.isEmpty()) {
+            for (FormDataBodyPart formDataBodyPart : processForms) {
+                processes.add(formDataBodyPart.getValueAs(Process.class));
             }
         }
+        return processes;
+    }
+
+    protected void submitEntities(String extensionName, String doAsUser, String jobName, List<Feed> feeds,
+                                  List<Process> processes, InputStream configStream) throws FalconException, IOException {
+        validateFeeds(feeds);
+        validateProcesses(processes);
+        List<String> feedNames = new ArrayList<>();
+        List<String> processNames = new ArrayList<>();
+        for (Feed feed : feeds) {
+            submitInternal(feed, doAsUser);
+            feedNames.add(feed.getName());
+        }
+        for (Process process: processes) {
+            submitInternal(process, doAsUser);
+            processNames.add(process.getName());
+        }
+
         ExtensionMetaStore metaStore = ExtensionStore.get().getMetaStore();
         byte[] configBytes = null;
         if (configStream != null) {
             configBytes = IOUtils.toByteArray(configStream);
         }
-        metaStore.storeExtensionJob(jobName, extensionName, feeds, processes, configBytes);
+        metaStore.storeExtensionJob(jobName, extensionName, feedNames, processNames, configBytes);
     }
 
-    private List<Entity> getEntityList(String extensionName, List<Entity> entities, InputStream config)
-        throws FalconException, IOException {
-        ExtensionType extensionType = getExtensionType(extensionName);
-        if (ExtensionType.TRUSTED.equals(extensionType)) {
-            entities = generateEntities(extensionName, config);
+    protected void scheduleEntities(Pair<List<Feed>, List<Process>> entityPairs) throws FalconException,
+            AuthorizationException {
+        for (Feed feed: entityPairs.first) {
+            scheduleInternal(EntityType.FEED.name(), feed.getName(), null, null);
+        }
+        for (Process process: entityPairs.second) {
+            scheduleInternal(EntityType.PROCESS.name(), process.getName(), null, null);
+        }
+    }
+
+
+    private void validateFeeds(List<Feed> feeds) throws FalconException {
```

Comment:
validateFeeds and validateEntities can be clubbed into single method, with type as an argument.